### ISSUE TITLE
[2026 Ambassador Application] Mu Xiaohui

### ISSUE THREAD

**Issue by Flink-ddd** [pytorch-fdn/ambassador-program#352]
### Application Type

Self-nomination

### Please confirm the following:

- [x] The nominee is 18 years of age or older.
- [x] The nominee agrees to follow the [PyTorch Foundation Code of Conduct](https://pytorch.org/code-of-conduct/).
- [x] The nominee agrees to comply with the [Linux Foundation Antitrust Policy](https://www.linuxfoundation.org/legal/antitrust-policy).
- [x] The nominee has active involvement in PyTorch Foundation projects or communities.
- [x] I confirm that this nominee has not already been submitted for the 2026 Ambassador cycle.

### Nominee Full Name

Mu Xiaohui

### Affiliation / Organization

RL-Align

### Nominee Email

vensenmu@gmail.com

### Current Location

Singapore

### Years of Community Involvement

1–2 years

### Which platforms or profiles would you like to share?

- [x] GitHub
- [ ] GitLab
- [x] LinkedIn
- [ ] X / Twitter
- [ ] Personal Website / Portfolio
- [ ] Other

### GitHub Profile

https://github.com/Flink-ddd

### LinkedIn Profile

https://www.linkedin.com/in/vensen/

### Personal Website / Portfolio

https://flink-ddd.github.io/

### Which PyTorch Foundation projects is the nominee familiar with?

- [x] PyTorch
- [x] vLLM
- [x] DeepSpeed
- [ ] Ray
- [ ] Helion
- [ ] Safetensors
- [ ] Other

### How has the nominee contributed to the community?

- [x] Open source contributions
- [x] Technical content creation
- [x] Community leadership
- [x] Events or workshops
- [ ] Research or academic work
- [x] Mentorship
- [ ] University or student outreach
- [ ] Translation or localization efforts
- [ ] Other

### Please describe the nominee’s contributions to PyTorch Foundation projects and communities

As an engineer specializing in AI Infrastructure and low-level optimization (CUDA/ROCm), I am dedicated to making deep technical contributions to PyTorch and its thriving ecosystem (vLLM, DeepSpeed, PyTorch). Building on these code-level contributions, I provide hardcore, source-code-level tutorials and career mentorship for the bilingual (English/Chinese) developer community. My core contributions span the following three dimensions:

1. Technical Contributions & Open Source Leadership

PyTorch Core Involvement: I actively contribute to PyTorch's low-level memory management. For example, I initiated and led the RFC discussion on c10::cuda::CUDACachingAllocator (Issue #174227) to address the synchronization gap between external physical memory releases (via VMM) and PyTorch's internal accounting, a critical feature for memory monitoring in LLM inference. I am a consistent contributor to the PyTorch repository.

Leading Next-Gen RL Post-Training Infrastructure (RL-Align/RL-Kernel): As the founder and core maintainer of RL-Kernel, I initiated this project to address the extreme memory bottlenecks (the $O(G \cdot L \cdot V)$ explosion) and sequence-padding waste in DeepSeek-style Large-Scale RL (GRPO/RLHF). It is an out-of-the-box, hardware-agnostic (CUDA, AMD ROCm, and domestic silicon) micro-level operator library. Crucially, I am actively building an open-source collective around this project to drive impact across three core pillars:

- Empowering Researchers: Providing plug-and-play kernel dispatching so algorithm researchers can execute GRPO training natively without Out-Of-Memory crashes.
- Solving Industry Pain Points: Delivering enterprise-grade solutions for the severe train-inference numerical divergence between rollout engines like vLLM, SGLang and training engines like Megatron, FSDP, DeepSpeed.
- Incubating Kernel Developers: Sourcing real-world industrial needs and mentoring passionate developers in our Discord community to write custom fused operators (e.g., fused_logp), guiding them to become impactful AI Infra contributors.

Driving Top-Tier Ecosystems (vLLM, vLLM-omni & DeepSpeed): I am a core contributor to vLLM, vLLM-omni, and DeepSpeed, focusing on LLM inference acceleration and distributed training. I spearheaded the development of the highly impactful Sleep Mode feature and its corresponding Ack Protocol (PR #2022) for vLLM-omni, and co-authored a deep-dive technical article on the official vLLM blog.

Low-Level Kernel Optimization: As a contributor to NVIDIA/cutlass, I continuously help build the foundational CUDA blocks that power the PyTorch acceleration stack.

2. Content Creation & Technical Evangelism

Source-Code-Level Medium Publications: Under the pen name Vensen, I write high-quality English technical blogs deconstructing the source code of DeepSpeed ZeRO-1/2/3 and vLLM's Sleep Mode. These articles bridge the gap between academic papers and PyTorch/CUDA source code, helping global developers understand complex AI Infra systems.

Video Evangelism (Bilibili): I consistently publish AI Infra video content for the Chinese-speaking community. My video "Deconstructing the 6 Core Directions of AI Infra" has been highly praised for guiding beginners. I also create comprehensive, step-by-step video tutorials ("Building a vLLM dev environment from scratch on RunPod") to significantly lower the barrier to entry for the PyTorch ecosystem. also, Invited by NVIDIA engineers, I recorded a walkthrough video about the vLLM architecture. The video is nearly two hours long and has been viewed over 1600 times.

3. Community Mentorship & Real-World Impact

Hands-On Developer Community: I run a dedicated developer mastermind group focused on AI Infra. Instead of just sharing knowledge, I emphasize practical conversion by designing specific learning tasks (source code compilation, environment setup, operator development) and providing 1-on-1 guidance.

Career Empowerment: Through my mentorship, community members have successfully transitioned into the field, landing core High-Performance Computing / AI Infra offers at top-tier tech companies like Tencent. This closed loop—from learning open-source PyTorch knowledge to securing top industry roles—is my proudest community achievement.

Measurable Impact
Community Growth: Built and manage a highly engaged AI Infra developer community of 200 active members, currently expanding into a second group due to high demand.

Community Reach: 4,300+ views on a single AI Infra foundational video on Bilibili, sparking dozens of high-quality technical discussions and guiding newcomers.

Knowledge Distribution: Consistent publication of deep-dive articles on Medium, precisely targeting professional developers focused on LLM optimization and PyTorch internals.

Mentorship Success: Directly mentored community members to secure core AI Infra roles at top-tier tech companies (e.g., Tencent), proving the real-world value of my educational initiatives.

<img width="897" height="879" alt="Image" src="https://github.com/user-attachments/assets/80108b35-b299-49a8-b9ad-9d1c33643f4d" />

Ecosystem Code Impact: Dozens of merged Pull Requests across PyTorch, DeepSpeed, vLLM, and CUTLASS, driving the iteration of foundational infrastructure relied upon by hundreds of thousands of developers globally.

Supporting Links
GitHub Repositories & Pull Requests

PyTorch RFC/Issue (Physical memory accounting for VMM): [Issue #174227](https://github.com/pytorch/pytorch/issues/174227)

PyTorch Code Contributions: [PyTorch PRs by Flink-ddd](https://github.com/pytorch/pytorch/pulls?q=is%3Apr+author%3AFlink-ddd+is%3Aclosed)

RL-Kernel Architecture & Core Issues: [RL-Kernel Issues by Flink-ddd](https://github.com/RL-Align/RL-Kernel/issues?q=is%3Aissue%20state%3Aopen%20author%3AFlink-ddd)

RL-Align/RL-Kernel Founders and members: [RL-Kernel PRs by Flink-ddd](https://github.com/RL-Align/RL-Kernel/pulls?q=is%3Apr+author%3AFlink-ddd+is%3Aclosed)

DeepSpeed Code Contributions: [DeepSpeed PRs by Flink-ddd](https://github.com/deepspeedai/DeepSpeed/pulls?q=is%3Apr+author%3AFlink-ddd+is%3Aclosed)

vLLM Code Contributions: [vLLM PRs by Flink-ddd](https://github.com/vllm-project/vllm/pulls?q=is%3Apr+author%3AFlink-ddd+is%3Aclosed)

CUTLASS Code Contributions: [CUTLASS PRs by Flink-ddd](https://github.com/NVIDIA/cutlass/pulls/Flink-ddd)

Blogs & Tutorials

Official vLLM Blog (Sleep Mode co-author: Vensen Mu): [vLLM Blog - Sleep Mode](https://vllm.ai/blog/2025-10-26-sleep-mode)

Medium Technical Column (Deconstructing DeepSpeed & vLLM): [Medium @vensenmu](https://medium.com/@vensenmu)

Videos & Live Streams

Bilibili Educational Video ("Deconstructing the 6 Core Directions of AI Infra"): https://b23.tv/WycFjVZ

<img width="585" height="1266" alt="Image" src="https://github.com/user-attachments/assets/8f68ba67-5527-4353-b0ef-a40bfcaae485" />
<img width="585" height="1266" alt="Image" src="https://github.com/user-attachments/assets/48fa2814-6274-46f7-ac2b-ac18d4335303" />

### Why does the nominee want to become a PyTorch Foundation Ambassador?

As an engineer spending my days hacking on PyTorch's CUDACachingAllocator, cutlass's cuteDSL and ROCm's  aiter and flyDSL and writing custom kernels for vLLM, I know exactly how intimidating the bottom layer of the AI stack can be. Most developers stop at calling torch.nn.Linear, completely terrified of C++, CUDA, and ROCm. When I transitioned into AI Infra, I hit that same wall.

I want to become an Ambassador because I love taking the pain out of that learning process. Running my AI Infra Collective online group, the most rewarding moment isn't just merging my own PRs, it’s watching a community member using that exact knowledge to land a core AI Infra offer at Tencent. We have also established a CUDA development group to combine CUDA, ROCm, Triton, and PyTorch native to develop a kernel integration platform. This will be released later and given back to the community.

I want to leverage the Foundation's platform to scale this impact. My motivation is to turn passionate PyTorch users into hardcore, confident system contributors who aren't afraid to dive deep into the source code.

### How would this nominee contribute as a PyTorch Foundation Ambassador?

If selected, my contributions won't just be about hosting general networking events. I will focus strictly on the hardcore engineering side of the PyTorch ecosystem, executing on three specific fronts:

1. Source-Code-Level Content Over High-Level Tutorials
The community has enough "How to train a basic model" tutorials. What’s missing is deep, accessible content on how PyTorch actually works under the hood. I will expand my Source-Code Deconstruction series on Medium and Bilibili. I will guide developers line-by-line through PyTorch internals, custom CUDA/Triton operator integration, and memory management, acting as a technical bridge for the English and Chinese-speaking communities.

2. Pushing PyTorch for the Extreme LLM Era
Because I actively build in the downstream LLM ecosystem (vLLM, DeepSpeed), I see exactly where PyTorch hits bottlenecks during massive distributed inference and training. I will actively drive technical discussions and upstream RFCs, like my current push to fix physical memory accounting for VMM (Issue #174227) to ensure PyTorch evolves perfectly alongside these critical acceleration frameworks.

3. Task-Driven Mentorship via the AI Infra Collective
Instead of passive lectures, I will scale my online developer community through action. I assign my members real-world, gritty engineering tasks: compiling PyTorch from source, tracing a memory leak, or writing a custom kernel. My concrete goal as an Ambassador is to mentor a new wave of developers to successfully merge their first PyTorch or vLLM Pull Request, and equip them with the actual combat skills needed to secure AI Infra jobs in the industry.

### Ambassador Focus Areas

- [x] Community Building
- [x] Technical Education
- [x] Open Source Contributions
- [ ] Research & Academia
- [x] Developer Advocacy
- [x] Mentorship
- [x] Events & Workshops
- [x] Content Creation
- [x] Regional Community Growth
- [ ] Other

### Primary Community Region

Asia

### Primary Country During Ambassador Term

Singapore and Malaysia

### Additional Information (Optional)

Moving forward, my technical roadmap includes collaborating more closely with PyTorch core maintainers on the compiler ecosystem. Specifically, I plan to dedicate more time to tackling open issues and contributing optimizations for PyTorch Dynamo and Inductor. Given my background in low-level kernel optimization and inference acceleration, I believe driving improvements in the torch.compile backend is critical for the next generation of AI systems, and I am highly committed to contributing my engineering bandwidth to this specific area during my Ambassador term.
